aya/src/obj/mod.rs

```diff
@@ -181,7 +210,7 @@ impl FromStr for ProgramSection {
 }
 
 impl Object {
-    pub(crate) fn parse(data: &[u8]) -> Result<Object, BpfError> {
+    pub(crate) fn parse(data: &[u8], globals: HashMap<&str, &[u8]>) -> Result<Object, BpfError> {
```

Comment:
Ok after realising that this is a "big diff" because of this change, I'm even more convinced that it should be done in a separate step.

Please revert parse() and parse_map() as they were before. Then add a Object::patch_data_maps(globals) method that does:

```
for (_, map) in self.maps.iter_mut() {
	patch_data(map.data, globals, self.symbols_by_index)?;
}
```

Reply:
sort of done